pkg/config/entity.go

```diff
@@ -0,0 +1,41 @@
+package config
+
+import (
+	"fmt"
+	"reflect"
+	"time"
+)
+
+const (
+	tblKeys      string = "config_keys"
+	tblValues    string = "config_values"
+	tblCache     string = "config_cache"
+	tblCacheDeps string = "config_cachedeps"
+)
+
+//convertTimestampToTime is converting the value of timestamp db-column to a Time instance
+func convertTimestampToTime(value interface{}) (interface{}, error) {
+	if reflect.TypeOf(value).Kind() == reflect.String {
+		layout := "2006-01-02 15:04:05" //see https://golang.org/src/time/format.go
+		return time.Parse(layout, value.(string))
+	}
+	if time, ok := value.(time.Time); ok {
+		return time, nil
+	}
+	return nil, fmt.Errorf("Failed to convert value '%s' (kind: %s) for field 'Created' to Time struct",
+		value, reflect.TypeOf(value).Kind())
+}
+
+func requireValidBucketName(value interface{}) (interface{}, error) {
+	bucketName := fmt.Sprintf("%s", value)
```

Comment:
Is the empty string a valid bucket name?

Reply:
no it isn't, but the validation should only run when the bucket string is defined. Marshaller are also executed in cases when an empty entity is provided (which is valid in cases of a delete-query, like here: https://github.com/kyma-incubator/reconciler/pull/2/files#diff-dae7efcf1d65d1b2a95798b43d9e45f920597f0889bd70a1e68fdf5a2e944909R70)

The validation for non-empty strings happens already via the annotation (see https://github.com/kyma-incubator/reconciler/pull/2/files#diff-2a76185f573c76ec3b23aff0fe2057dae5d020b1866407aadec53fd4a0fbef91R14). I'm actually abusing here the marshaller to define another validation for the value. A marshaller should normally only convert data-types from Go-Types to a format the DB understands and only fail if the value is not convertible. Here I let it fail when the provided value is not fitting to my expectations.